### PR TITLE
Chapter 3: Mostly Static Pages

### PR DESCRIPTION
**Pull request by ainulfarizah**
ready

### CODE REVIEW

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
Screenshot from 2021-08-24 14-14-03
how to solve this error?

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+ before_action :switch_locale
 
 def switch_locale(&action)
-  locale = current_user.try(:locale || I18n.default_locale
-  I18n.with_locale(locale, &action)
+   locale = current_user.try(:locale) || I18n.default_locale
+   I18n.with_locale(locale, &action)
 end
 
 def switch_locale(&action)
```
Comment (huongnt-2545) [inline app/controllers/application_controller.rb]: duplicate function with function in line 4th

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,23 +1,24 @@
 class ApplicationController < ActionController::Base
-around_action :switch_locale
+ before_action :switch_locale
```
Comment (huongnt-2545) [inline app/controllers/application_controller.rb]: indent = 2 spaces
Fix the same error in this file
```suggestion
  before_action :switch_locale
```

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,23 +1,24 @@
 class ApplicationController < ActionController::Base
```
Comment (thaodtd-0665) [inline app/controllers/application_controller.rb:1]:
```suggestion
class ApplicationController < ActionController::Base
  before_action :switch_locale

  private
  
  def switch_locale
    I18n.locale = params[:locale] || I18n.default_locale
  end

  def default_url_options
    { locale: I18n.locale }
  end
end
```

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
> ready

**Comment by ainulfarizah** [comment]
Screenshot from 2021-08-24 16-27-54
how to solve this?

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
Screenshot from 2021-08-24 17-34-40
how to change txt file to scss file?

**Review by huongnt-2545 — COMMENTED** [review]
Some comments are not applied, please check and apply to resolve errors

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
image
may i know how to solve this

**Comment by ainulfarizah** [comment]
image
why is my text small?

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+padding-top: 60px;
+}
+section {
+overflow: auto;
+}
+textarea {
+resize: vertical;
+}
+.center {
+text-align: center;
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]:
```suggestion
.center {
  text-align: center;
```

Comment (ngocvt-0484) [reply app/assets/stylesheets/custom.scss]: wrong indent

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+.center h1 {
+margin-bottom: 10px;
+}
+
+/* header */
+#logo {
+float: left;
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  float: left;
```

File: exit [hunk 6]
```diff
@@ -0,0 +1,38 @@
+[33mcommit 7be2791b4c0084159fe6a3031aa92900b8f79175[m[33m ([m[1;36mHEAD -> [m[1;32mchapter_3[m[33m, [m[1;31morigin/chapter_3[m[33m)[m
+Author: ainulfarizah <ainul.farizah@gmail.com>
```
Comment (ngocvt-0484) [inline exit]: delete this file

**Comment by ainulfarizah** [comment]
ready

File: app/models/usesr.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
```
Comment (thaodtd-0665) [inline app/models/usesr.rb]: remove this file, name "usesr.rb" incorrect

File: app/controllers/home_controller.rb [hunk 8]
```diff
@@ -1,5 +1,2 @@
 class HomeController < ApplicationController
```
Comment (ghost) [inline app/controllers/home_controller.rb]: delete this file if you don't need it

File: config/routes.rb [hunk 9]
```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
-  get '/:locale/' => 'dashboard#index'
-  root to: "home#index"
-  scope "/:locale", locale: /en|nl/ do
-    resources :books
+  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
+    get 'home/index'
```
Comment (thaodtd-0665) [inline config/routes.rb]: remove this line, recommend use 'static_pages#home' for home page
```suggestion
```

File: app/controllers/home_controller.rb (hunk 8 above)
Comment (thaodtd-0665) [inline app/controllers/home_controller.rb]: Recommend using StaticPagesController, so remove this file

File: app/models/usesr.rb (hunk 7 above)
Comment (ghost) [inline app/models/usesr.rb]: delete this file if you don't use it

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ainulfarizah** [comment]
ready

**Comment by ainulfarizah** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**